src/shared/utils/networks.ts

```diff
@@ -0,0 +1,47 @@
+import config from './config'
+
+interface Network {
+  network: string
+  port?: number
+  entry: string
+  unls: string[]
+}
+
+// put the UNL you want to prioritize at the front
+const mainMainnetUnls = ['vl.ripple.com', 'vl.xrplf.org', 'vl.coil.com']
+let mainnetUnls
+if (config.mainnet_unl == null) {
+  mainnetUnls = mainMainnetUnls
+} else {
+  mainnetUnls = [config.mainnet_unl]
+  mainMainnetUnls.forEach((unl) => {
+    if (unl !== config.mainnet_unl) {
+      mainnetUnls.push(unl)
+    }
+  })
+}
```

Comment:
I think this can be simplified using Sets.
```suggestion
const mainMainnetUnls = new Set(['vl.ripple.com', 'vl.xrplf.org', 'vl.coil.com'])
if(config.mainnet_unl != null) {
  mainMainnetUnls.add(config.mainnet_unl)
}
mainnetUnls = Array.from(mainMainnetUnls)
```

Reply:
That doesn't solve the problem of wanting `config.mainnet_unl` to be at the front of the array if it's not `null`

Reply:
Ah right :)